crates/chain/src/spk_txout_index.rs

```diff
@@ -301,9 +305,11 @@ impl<I: Clone + Ord> SpkTxOutIndex<I> {
     /// for calling [`sent_and_received`] and subtracting sent from received.
     ///
     /// [`sent_and_received`]: Self::sent_and_received
-    pub fn net_value(&self, tx: &Transaction, range: impl RangeBounds<I>) -> i64 {
+    pub fn net_value(&self, tx: &Transaction, range: impl RangeBounds<I>) -> SignedAmount {
         let (sent, received) = self.sent_and_received(tx, range);
-        received as i64 - sent as i64
+        (received - sent)
+            .to_signed()
+            .expect("Amount conversion to SignedAmount error")
```

Comment:
I think you want to do `to_signed` on each Amount before doing the subtraction.

Reply:
Yes, you're right, fixed it.